domain/src/user.rs

```diff
@@ -150,3 +153,64 @@ pub async fn delete(db: &DatabaseConnection, user_id: Id) -> Result<(), Error> {
 
     Ok(())
 }
+
+pub async fn create_by_organization(
+    db: &DatabaseConnection,
+    config: &Config,
+    organization_id: Id,
+    user_model: users::Model,
+) -> Result<users::Model, Error> {
+    // Create the user first using the entity_api function
+    let new_user =
+        entity_api::user::create_by_organization(db, organization_id, user_model).await?;
+
+    // Send welcome email in the background
+    send_welcome_email(config, &new_user).await?;
+
+    Ok(new_user)
+}
+
+/// Send a welcome email to a newly created user
+async fn send_welcome_email(config: &Config, user: &users::Model) -> Result<(), Error> {
+    log::info!(
+        "Initiating welcome email for user: {} ({})",
+        user.email,
+        user.id
+    );
+
+    let mailersend_client = MailerSendClient::new(config).await?;
```

Comment:
Something to consider. Obviously not a problem today, but could become a fairly big memory hog when it doesn't need to be, especially since you're spawning sub-threads in the mail sender anyway.

  **Current: Creating new client for each email**
```rust
  let mailersend_client = MailerSendClient::new(config).await?;
```

  **Better: Client should be created once and reused**
```rust
  // In AppState or similar
  pub struct EmailService {
      client: Arc<MailerSendClient>,
  }
```

Reply:
yeah didn't think of this at all. You're much more experienced with this type of thing so, potentially naive question I would have is:
if we consider this the same referenced object in memory, will it continue to spawn new threads every time it's invoked? 
If yes: does it make a difference that the base object in memory is copied or not in that case?
If no: will we bottleneck email dispatches?

Or will `MailerSendClient` be an object in only a single instance of the app in which case emails will be invoked sequentially because the single user using that instance of the app is making sequential actions even though the emails themselves are dispatched asynchronously relative to the calling thread? 
More for my academic understanding than anything